Contest_API.md

```diff
@@ -1226,6 +1226,31 @@ Returned data:
 {"id":"nicky","type":"admin"}
 ```
 
+### Account
+
+Provides information on which account the client represents within a given contest.
+
+The following endpoint is associated with an account:
+
+| Endpoint                 | Mime-type        | Required? | Description
+| :----------------------- | :--------------- | :-------- | :----------
+| `/contests/<id>/account` | application/json | yes       | JSON object representing a single account with properties as defined in the [Accounts](#accounts) table above.
+
+This endpoint exists in the API so that the clients can tell which account (and hence which person or team) they are logged in as. It is not
+expected to exist in a contest archive, and will fail with a 401 error code if the client is not authenticated.
```

Comment:
401 is the one where the server is expected to send a WWW-Authenticate header with a challenge, which the client should use to show a login prompt (see [here](https://www.webfx.com/web-development/glossary/http-status-codes/what-is-a-401-status-code/)). That doesn't seem like what we want? Maybe we want [403](https://www.webfx.com/web-development/glossary/http-status-codes/what-is-a-403-status-code/)?

Reply:
That doesn't feel perfectly right either.

If this was a list we could return an empty list... but it isn't, and it shouldn't be.

Reply:
Yeah I can't seem to find a better HTTP status code. But 401 is definitely not the correct one. Or if it is, we NEED to send this strange WWW-Authenticate header. Which seems wrong.

Reply:
I was waffling between 401 and 403 initially, switched it to 403 now. Fredrik, Nicky and I can't find anything better, so please approve or provide another suggestion. :)

Reply:
- 401 is definitely wrong.
- I don't think 403 is quite right either.
- 404 is kinda technically correct.
- An actual empty result (i.e. `""`) is also kinda technically correct.

If this was an array then returning an empty array (`"[]"`) would be nicer and clearer, but it isn't an array, and I don't think we should make it one just for this.

I think the empty result would be the best choice.

Reply:
But that’s not valid json is it? Then it should return `null`.
What makes 403 not right? As far as I know it means “with your current credentials you do not have permission to view this entity” which kinda is true?

Reply:
I agree with Nicky that it'd have to return `null`. An empty array or "" are also valid JSON, but then you need to document that two different JSON types may be returned, which is awkward and harder for clients to deal with.
IMHO HTTP status code is the most straightforward to document and for clients to handle.

I have seen sites use 404 for things you need to log in to see, to avoid giving out any information to unauthenticated users. If the server didn't support unauthenticated access and is returning 404 for /contests then I'd assume it would do the same here, but if /contests exists for the user then 404 seems wrong to me; you know this endpoint exists.

An unauthenticated user asking what account they have is an edge case though. I disagree with `[]` or `""`, but if Fredrik feels strongly about 404 or `null` I'd rather change, merge, and discuss on the next call vs delaying getting the rest in. :)

Reply:
> But that’s not valid json is it?

No, it seems it isn't. I don't know why I thought it would be. (Maybe because it totally should be 😄).

> Then it should return `null`.

But it also really shouldn't. `null` isn't a JSON object with properties as we have defined.

 Now, if everything is optional, then an empty *object* would be correct (i.e. `"{}"`).

> What makes 403 not right? As far as I know it means “with your current credentials you do not have permission to view this entity” which kinda is true?

Kinda, a bit, maybe. I would argue that 403 implies that there exists an object here, but you are not allowed to see it. But the object in fact does not exist, which would imply 404. If you *do* log i, you will get some *other* object, not this one. Which one depends on what you authenticate as.

So, let me update my stance:
- 401 is definitely wrong.
- *I* think 403 is somewhat (and very technically) wrong.
- 404 is kinda technically correct.
- An empty result `""` is definitely wrong, since it's not even valid JSON.
- An empty object `"{}"` would be correct if everything is optional, which it isn't, and I don't think we want to make it.
- An empty array `"[]"` would be correct if we changed the expected return to be an array, but I really don't think we want to do that.
- `null` would be correct only if we explicitly said that the result could be "a JSON account object or null". Which, I guess would be fine.

So... I think 404 is best, followed by `null`?

Yet another option (because clearly we don't have enough) would be to define an implicit account for the unauthenticated that always has to be there. That would of course be fine, but I think I prefer 404 and `null` over that?

Reply:
I think using 404 or 403 is better than returning weird JSON responses (either `null` or `{}`) that don't match the normal specification of this endpoint. If we're to return an exceptional state (no account), I think it makes most sense to do so using an error code.

Reply:
Ok, I updated to 404.